src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp

```diff
@@ -104,7 +109,103 @@ class boss_magmadar : public CreatureScript
         }
 };
 
+// Smoldering animation is an hack, Serrated Bites timer may be wrong
+// The original smoldering aura should increase crit chance to 100% and play dead animation
+class npc_magmadar_core_hound : public CreatureScript
+{
+public:
+	npc_magmadar_core_hound() : CreatureScript("npc_magmadar_core_hound") { }
+
+	struct npc_magmadar_core_houndAI : public CreatureAI
+	{
+		npc_magmadar_core_houndAI(Creature* creature) : CreatureAI(creature)
+		{
+		}
+
+		EventMap events;
+		bool smoldering;
+		std::list<Creature *> hounds;
+
+		void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damagetype*/, SpellSchoolMask /*damageSchoolMask*/) 
+		{
+			if (me->HealthBelowPctDamaged(0, damage))
+			{
+				if (!smoldering)
+				{
+					events.ScheduleEvent(EVENT_IGNITE, 10000);
+					me->SetHealth(1);
+					me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_STUNNED|UNIT_FLAG_PACIFIED);
+					me->AddAura(57626, me); // feign death animation, this is an hack even if the result is good
+					Talk(EMOTE_SMOLDERING);
+				}
+				damage = 0;
+				smoldering = true;
+			}
+		}
+
+		void Reset()
+		{
+
+		}
+
+		void EnterCombat(Unit* victim)
+		{
+			events.ScheduleEvent(EVENT_SERRATED_BITE, 10000); // timer may be wrong
+			smoldering = false;
+		}
+
+		void UpdateAI(uint32 diff)
+		{
+			if (!UpdateVictim())
+				return;
+
+			events.Update(diff);
+
+			while (uint32 eventId = events.ExecuteEvent())
+			{
+				switch (eventId)
+				{
+				case EVENT_SERRATED_BITE:
+					if (!me->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PACIFIED))
+						DoCast(me->GetVictim(), SPELL_SERRATED_BITE);
+					events.ScheduleEvent(EVENT_SERRATED_BITE, 10000); // again, timer may be wrong
+					break;
+				case EVENT_IGNITE:
+					me->GetCreaturesWithEntryInRange(hounds, 100, NPC_CORE_HOUND);
+					for (Creature * i : hounds)
+					{
+						if (i && i->IsAlive() && i->IsInCombat() && !i->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PACIFIED))
+						{
+							me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_STUNNED|UNIT_FLAG_PACIFIED);
+							me->RemoveAura(me->GetAura(57626), AURA_REMOVE_BY_DEFAULT);
+							me->SetFullHealth();
+							smoldering = false;
+							Talk(EMOTE_IGNITE);
+							return;
+						}
+					}
+					Unit::Kill(me, me);
+					break;
+				default:
+					break;
+				}
+			}
+
+			if (me->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PACIFIED))
+				return;
+
+			DoMeleeAttackIfReady();
+		}
+	};
+
+	CreatureAI* GetAI(Creature* creature) const
+	{
+		return new npc_magmadar_core_houndAI(creature);
+	}
+};
+
 void AddSC_boss_magmadar()
 {
     new boss_magmadar();
+	new npc_magmadar_core_hound();
```

Comment:
please fix indentation

Reply:
Just saw your comments at the top here after I posted on the issue.